### DISCUSSION TITLE
MAUI: From Cross-Platform Paradise Promises to Buggy Purgatory?

### DISCUSSION THREAD

**Discussion by ViperTricks** [dotnet/maui#19525] [Show and tell]
Listen up, MAUI community! Buckle up because I'm about to unleash a tale of epic proportions - a saga of shattered promises, buggy nightmares, and the audacity of a billion-dollar company peddling a broken framework like it's the next big thing.

I, a battle-hardened full-stack .NET dev, was lured into the MAUI trap by its siren song of cross-platform paradise. But instead of palm trees and piña coladas, I found myself knee-deep in a swamp of bugs.

Remember that thread on Reddit, the one titled "The MAUI Team is a Mess"?(https://www.reddit.com/r/dotnet/comments/18is0ik/the_maui_team_is_a_mess/). I PM'd the broken soul behind it, eager to dive deeper into the madness. He, bless his kind, shared his app with me. And oh boy, let me tell you, it was an eye-opener.

Buttons stuck in limbo, images refusing to budge, a CollectionView so dysfunctional it belonged in a museum of tech disasters. Every click was a gamble, every scroll a prayer. But the real cherry on top of this buggy sundae? The Android and iOS targets spontaneously combusting without warning! Talk about adding insult to injury.

And then came the "defense" from the MAUI team (https://www.reddit.com/r/dotnetMAUI/comments/18lxeud/a_bit_of_maui_love_again/). Oh, the irony! They hide behind their "small team" excuse, like a flimsy shield against a barrage of valid criticism. Small team? You're nestled in the bosom of a BILLION-DOLLAR company! Where's the rest of that cash going, huh? Funding Bing ads while your framework crumbles like a stale croissant?

Three versions, four years, and still, the most basic features are riddled with holes. You call that progress? Your "proud" boasts ring hollower than a deflated basketball. Show me results, not self-congratulatory pats on the back!

Don't even try to compare this glorified buggy beta to Flutter. They've got a mountain of resolved issues, not an open wound like yours. And your age is no excuse. You're just a reskin of Xamarin, dragging its baggage of bugs from a bygone era.

MAUI, you're not new, you're not innovative, you're just a frustrating disappointment. Take your "small team" excuse and shove it! We, the developers, deserve better. We won't be fooled again. This community needs a framework that works, not a never-ending stream of frustration.

So listen up, MAUI team! Get your act together, or prepare to be left in the dust by developers who refuse to waste their time on broken promises. This is a call to arms, a rallying cry for those who've been burned by your buggy mess.

**Comment by Vroomer** [comment]
So you have no real experience with MAUI?

**Comment by akhanalcs** [comment]
I still don't understand why MAUI is still a small team. Shouldn't MSFT put more resources into something whose scope is this big?

**Comment by mikkpaasuke** [comment]
An anonymous rant on reddit and a related one by someone with nothing to see on their Linked-In profile despite claiming to be 'battle hardened'. Which 'we' are you claiming to speak for, exactly?

I'm only using .NET MAUI for Android (internal company application), but - it's fine. Sure, there are some frustrations and issues, but nothing that stops me delivering critical functionality. The quality and reliability has improved noticeably over the ~12 months I've been developing this application, especially with the recent .NET 8 release (and upgrading to this only took several hours of messing about). How much time have you actually spent yourself on .MAUI development in a real-world situation? Not much, I'm guessing. 

Regarding the Xamarin history - I went through a similar situation developing a Xamarin for Mac application, which despite some rough patches early on has now been stable and commercially available for years. Now with .NET Maui, if I'm stuck I have years worth of Xamarin Forms SO posts/blogs/other material to wade through to find a solution, so this 'legacy' is for me at least an advantage as a solo developer.

Your direction of criticism at the .NET MAUI team is, I believe, extremely misplaced. Maybe they work for 'big MS', but they are fellow developers doing, I would like to believe, their best in not the easiest of circumstances. Perhaps a little more real-world experience will help you to have a more generous and empathetic perspective? Especially as you don't seem to have much invested in .NET MAUI to begin with.

**Comment by Xyncgas** [comment]
I can display bytes as image on browser, how am I going to do it using this framework
Is it way way difficult to implement a customer's ContextMenu on this framework than on a browser

**Reply by Stedy59** [reply]
byte[.] => MemoryStream => ImageSource
Done!

**Reply by Xyncgas** [reply]
Thanks
